lib/charms/vault_k8s/v0/vault_managers.py

```diff
@@ -504,3 +550,304 @@ def generate_vault_unit_certificate(
 def existing_certificate_is_self_signed(ca_certificate: Certificate) -> bool:
     """Return whether the certificate is a self signed certificate generated by the Vault charm."""
     return ca_certificate.common_name == VAULT_CA_SUBJECT
+
+
+class VaultAutounsealNaming:
+    """Computes the auto-unseal related values for a relation.
+
+    This class is used to compute the static details for a vault-autounseal
+    relation, such as the key name, policy name, and approle name. These values
+    are all based on the relation ID.
+
+    This class provides a central place to manage the naming conventions for
+    the auto-unseal functionality.
+    """
+
+    key_prefix: str = ""
+    policy_prefix: str = "charm-autounseal-"
+    approle_prefix: str = "charm-autounseal-"
+
+    @classmethod
+    def key_name(cls, relation_id) -> str:
+        """Return the key name for the relation."""
+        return f"{cls.key_prefix}{relation_id}"
+
+    @classmethod
+    def policy_name(cls, relation_id) -> str:
+        """Return the policy name for the relation."""
+        return f"{cls.policy_prefix}{relation_id}"
+
+    @classmethod
+    def approle_name(cls, relation_id) -> str:
```

Comment:
We're missing the input type

(Does one of our tool have an automatic checker for this?)

Reply:
https://github.com/canonical/vault-k8s-operator/pull/534/commits/c35ab15147e33882eac1d1a8e6ffd3a7a694ac6e